clingo/plan.sh

```diff
@@ -16,6 +16,8 @@ pkg_build_deps=(
 pkg_deps=(
   core/gcc-libs
   core/glibc
+  core/python
```

Comment:
Is Python necessarily needed? From a quick look at [how it builds](https://github.com/potassco/clingo/blob/master/.travis.yml) I think it does need Lua, but I'm unsure about Python

Reply:
@james-stocks You can test this by removing the `core/python` line from the plan, and then  building it.

For your reference, it builds fine with python. And fails as follows when it is removed:

```
[ 95%] Linking CXX executable ../../bin/gringo
/hab/pkgs/core/binutils/2.31.1/20190115003743/bin/ld.bfd.real: /hab/pkgs/core/lua/5.3.4/20190117191939/lib/liblua.a(loadlib.o): in function `lookforfunc':
(.text+0x4e4): undefined reference to `dlsym'
/hab/pkgs/core/binutils/2.31.1/20190115003743/bin/ld.bfd.real: (.text+0x546): undefined reference to `dlopen'
/hab/pkgs/core/binutils/2.31.1/20190115003743/bin/ld.bfd.real: (.text+0x5c9): undefined reference to `dlerror'
/hab/pkgs/core/binutils/2.31.1/20190115003743/bin/ld.bfd.real: (.text+0x5f1): undefined reference to `dlerror'
/hab/pkgs/core/binutils/2.31.1/20190115003743/bin/ld.bfd.real: /hab/pkgs/core/lua/5.3.4/20190117191939/lib/liblua.a(loadlib.o): in function `gctm':
(.text+0x7af): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
make[2]: *** [app/gringo/CMakeFiles/gringo.dir/build.make:87: bin/gringo] Error 1
make[2]: Leaving directory '/hab/cache/src/clingo-5.3.0/build'
make[1]: *** [CMakeFiles/Makefile2:704: app/gringo/CMakeFiles/gringo.dir/all] Error 2
make[1]: Leaving directory '/hab/cache/src/clingo-5.3.0/build'
make: *** [Makefile:141: all] Error 2
make: Leaving directory '/hab/cache/src/clingo-5.3.0/build'
   clingo: Build time: 3m32s
   clingo: Exiting on error
```

Reply:
Hi @james-stocks.  I've tested this also:  if I remove core/python from $pkg_deps, I get the same error you see above.

Interestingly, if I move the core/python from $pkg_deps to $pkg_build_deps the build succeeds but the tests fail proving the value of the tests.  ``hab studio run "./clingo/tests/test.sh $pkg_ident"`` fails as follows:

```bash
1..3
not ok 1 Version matches plan
# (in test file clingo/tests/test.bats, line 3)
#   `actual_version="$(hab pkg exec "${TEST_PKG_IDENT}" gringo --version | grep -E "^gringo version")"' failed
# /hab/pkgs/core/clingo/5.3.0/20190625142943/bin/gringo: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
not ok 2 gringo should work as expected for python
# (in test file clingo/tests/test.bats, line 9)
#   `actual=$( hab pkg exec core/clingo gringo --text ${TESTDIR}/fixtures/examples/gringo/sort/sort-py.lp ${TESTDIR}/fixtures/examples/gringo/sort/encoding.lp )' failed with status 127
# /hab/pkgs/core/clingo/5.3.0/20190625142943/bin/gringo: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
not ok 3 gringo should work as expected for lua
# (in test file clingo/tests/test.bats, line 16)
#   `actual=$( hab pkg exec core/clingo gringo --text ${TESTDIR}/fixtures/examples/gringo/sort/sort-lua.lp ${TESTDIR}/fixtures/examples/gringo/sort/encoding.lp )' failed with status 127
# /hab/pkgs/core/clingo/5.3.0/20190625142943/bin/gringo: error while loading shared libraries: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```